### PR TITLE
removes duplicate variable reference crash from pad_sequences

### PR DESCRIPTION
**Pull request by williamFalcon**
This closes issue #4381  .

Please let me know your views and suggest improvements if any.
One key thing I wasn't clear on was whether it needs to support tensors with more than 3 dimensions. If so, happy to make the changes.

### CODE REVIEW

**Review by apaszke — COMMENTED** [review]
Nice catch. This mostly looks good but:
1. We need to support more dimensions. You can leave the code as it was before and only keep the few lines that replace the select with if-else. (also, I find `:length` much clearer than `variable.size(0)`, so please use that).
2. There are a bunch of formatting changes in the doc strings. Please revert those.

**Comment by williamFalcon** [comment]
Done and pushed. Let me know if that looks good.

File: torch/nn/utils/rnn.py [hunk 1]
```diff
@@ -172,14 +172,12 @@ def pad_sequence(sequences, batch_first=False):
     # assuming trailing dimensions and type of all the Variables
     # in sequences are same and fetching those from sequences[0]
     max_size = sequences[0].size()
-    max_len, trailing_dims = max_size[0], max_size[1:]
+    max_len, trailing_dims = max_size[0], max_size[2:]
```
Comment (apaszke) [inline torch/nn/utils/rnn.py]: trailing dims should take everything starting from 1, not 2

**Review by apaszke — CHANGES_REQUESTED** [review]

**Review by apaszke — APPROVED** [review]
The error was tiny, so I fixed it for you

**Comment by soumith** [comment]
thanks @williamFalcon !

**Pull request merged**